example/lib/test_scenarios/peripheral_test_operations.dart

```diff
@@ -456,6 +492,223 @@ class PeripheralTestOperations {
     });
   }
 
+  Future<void> readDescriptorForPeripheral() async =>
+      _runWithErrorHandling(() async {
+        log("READ DESCRIPTOR FOR PERIPHERAL");
+        log("Reading value...");
+        Uint8List value = await peripheral
+            .readDescriptor(
+              SensorTagTemperatureUuids.temperatureService,
+              SensorTagTemperatureUuids.temperatureDataCharacteristic,
+              SensorTagTemperatureUuids
+                  .clientCharacteristicConfigurationDescriptor,
+            )
+            .then((descriptorWithValue) => descriptorWithValue.value);
+        log("Value $value read!");
+      });
+
+  Future<void> readDescriptorForService() async =>
+      _runWithErrorHandling(() async {
+        log("READ DESCRIPTOR FOR SERVICE");
+
+        log("Fetching service");
+        List<Service> services = await peripheral.services();
+        Service chosenService = services.firstWhere((elem) =>
+            elem.uuid ==
+            SensorTagTemperatureUuids.temperatureService.toLowerCase());
+
+        log("Reading value...");
+        Uint8List value = await chosenService
+            .readDescriptor(
+              SensorTagTemperatureUuids.temperatureDataCharacteristic,
+              SensorTagTemperatureUuids
+                  .clientCharacteristicConfigurationDescriptor,
+            )
+            .then((descriptorWithValue) => descriptorWithValue.value);
+        log("Value $value read!");
+      });
+
+  Future<void> readDescriptorForCharacteristic() async =>
+      _runWithErrorHandling(() async {
+        log("READ DESCRIPTOR FOR CHARACTERISTIC");
+
+        log("Fetching service");
+        List<Service> services = await peripheral.services();
+        Service chosenService = services.firstWhere((elem) =>
+            elem.uuid ==
+            SensorTagTemperatureUuids.temperatureService.toLowerCase());
+
+        log("Fetching characteristic");
+        List<Characteristic> temperatureCharacteristics =
+            await chosenService.characteristics();
+        Characteristic chosenCharacteristic = temperatureCharacteristics.first;
+
+        log("Reading value...");
+        Uint8List value = await chosenCharacteristic
+            .readDescriptor(
+              SensorTagTemperatureUuids
+                  .clientCharacteristicConfigurationDescriptor,
+            )
+            .then((descriptorWithValue) => descriptorWithValue.value);
+        log("Value $value read!");
+      });
+
+  Future<void> readDescriptor() async => _runWithErrorHandling(() async {
+        log("READ DESCRIPTOR FOR DESCRIPTOR");
```

Comment:
I couldn't find reading descriptor for descriptor in the code below. 🤔

Reply:
This is testing  `descriptor.read()`